### PR TITLE
[Hardware][Ascend] Keep HiFT vocoder on NPU

### PR DESCRIPTION
**Pull request by 00fish0**
## Purpose

The current Ascend workaround introduced by #5117 offloads the entire HiFT vocoder to CPU to avoid an ACL 507015 failure.

With synchronous NPU execution enabled, the failure was isolated to HiFT `SineGen2._f02sine`:

- input shape: `[1, 9, 24000]`
- output shape: `[1, 9, 50]`
- operation: `F.interpolate(..., scale_factor=1/480, mode="linear")`
- failure: AIVector UB-address exception / ACL 507015

Standalone STFT/ISTFT with the same waveform shape passed. Specifying an explicit output size still failed, while the reverse 480x linear upsample passed. This is consistent with Ascend/pytorch#150.

For an even integer downsampling scale and `align_corners=False`, each output coordinate lies exactly halfway between two input samples. Therefore, the linear interpolation can be replaced by selecting and averaging the two midpoint samples:

```python
left = scale // 2 - 1
right = scale // 2
output = (x[..., left::scale] + x[..., right::scale]) * 0.5
```

This PR:
- replaces only the failing HiFT linear downsample on Ascend;
- keeps the complete HiFT vocoder, including STFT/ISTFT, on NPU;
- removes per-chunk CPU offload and NPU/CPU host transfers;
- falls back to the original implementation for pulse mode, non-integer or
- odd scales, and non-divisible input lengths;
- does not change the CUDA or CPU execution paths;
- adds CPU unit tests for numerical equivalence, fallback validation, idempotency, and device placement.

### Unit Test

```
pytest -q tests/platforms/npu/test_step_audio2_token2wav.py
```
The tests cover:
- Exact equality with F.interpolate(..., mode="linear").
- Rejection of odd scales and non-divisible input lengths.
- Idempotent HiFT patching.
- Verification that the patch does not call hift.to("cpu").
- Exact SineGen2._f02sine output equivalence with a fixed random seed.

### Warm-request latency comparison

A sequential batch-size-1 A/B test was retained in the stopped validation containers. Both variants used the same deployment configuration and generated 48,000 audio frames (2.0 seconds at 24 kHz). The first request was excluded as a model/operator warm-up request.

| Metric | HiFT CPU offload | HiFT on NPU | Change |
|---|---:|---:|---:|
| Samples | 3 | 4 | — |
| Mean end-to-end latency | 2,811.489 ms | **1,521.942 ms** | -45.9% |
| End-to-end latency range | 2,693.248–2,875.531 ms | **1,516.086–1,528.440 ms** | — |
| Mean Stage 1 `vllm_ttft_ms` | 2,614.162 ms | **1,327.212 ms** | -49.2% |
| Mean Stage 0 `vllm_ttft_ms` | 76.931 ms | **75.629 ms** | approximately unchanged |
| Real-time factor | 1.406 | **0.761** | 1.85x faster |

Raw end-to-end samples:

- CPU offload: **`2693.248`, `2865.687`, `2875.531` ms**
- HiFT on NPU: **`1528.440`, `1518.255`, `1516.086`, `1524.987` ms**

### CODE REVIEW

**Comment by amy-why-3459** [comment]
Thanks for the careful root-cause isolation — this is a clear improvement over the previous whole-module HiFT CPU offload.

### Overall
The approach looks reasonable and mergeable: keep HiFT on NPU, and replace only the failing even-integer linear downsample in SineGen2._f02sine with the exact midpoint form. For even scale + length divisible by scale + align_corners=False, that rewrite is bitwise-equivalent to F.interpolate(..., mode="linear"), and avoiding NPU↔CPU copies explains the large latency win.

### Suggestions / concerns

**1. Fallback path can reintroduce ACL 507015 on NPU (high)**
In _f02sine_on_npu, unsupported cases (flag_for_pulse, odd/non-integer scale, non-divisible length) fall back to the original _f02sine, which still calls the broken linear1d kernel. Under the old CPU-offload workaround those edges were safe; with this PR they can crash again.
Suggestion: for fallback, temporarily run that path on CPU (or raise a clear error), instead of calling the original NPU implementation.

**2. Patched _f02sine does not mirror the causal branch (medium)**
Upstream _f02sine uses fixed rand_ini + nearest upsample when causal=True, but _f02sine_on_npu always uses the non-causal/linear path. If HiFTGenerator() is always non-causal in this stack, please state that explicitly (or assert it). Otherwise please mirror the causal branch too.

**3. Hard dependency on hift.m_source.l_sin_gen (low)**
Consider a more defensive lookup / clearer error if the flashcosyvoice HiFT layout changes.

**4. Test coverage gap (medium)**
The new CPU unit tests are great for numerical equivalence / idempotency / “no hift.to("cpu")”. They don’t prove STFT/ISTFT stability on Ascend after removing CPU offload. A short NPU TTS smoke (or listing validated hardware/CANN versions in the PR description) would make this much safer to land.

**5. Docs follow-up (low, optional)**
Ascend deployment notes / FAQ that still say “HiFT is fully CPU-offloaded” will be stale after this lands.

**Review by amy-why-3459 — APPROVED** [review]

**Comment by amy-why-3459** [comment]
Thanks, LGTM.
